src/explore-education-statistics-admin/src/modules/find-statistics/components/EditableKeyStatTile.tsx

```diff
@@ -0,0 +1,200 @@
+import Button from '@common/components/Button';
+import Details from '@common/components/Details';
+import { Form, FormFieldTextInput, Formik } from '@common/components/form';
+import FormFieldTextArea from '@common/components/form/FormFieldTextArea';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import formatPretty from '@common/lib/utils/number/formatPretty';
+import KeyStatTile, {
+  KeyStatConfig,
+  KeyStatProps,
+} from '@common/modules/find-statistics/components/KeyStatTile';
+import styles from '@common/modules/find-statistics/components/SummaryRenderer.module.scss';
+import DataBlockService, {
+  DataBlockResponse,
+} from '@common/services/dataBlockService';
+import { Summary } from '@common/services/publicationService';
+import { FormikProps } from 'formik';
+import React, { useEffect, useState } from 'react';
+
+type KeyStatsFormValues = Omit<Summary, 'dataKeys'>;
+
+interface EditableKeyStatProps extends KeyStatProps {
+  isEditing?: boolean;
+  onRemove?: () => void;
+  onSubmit: (values: KeyStatsFormValues) => void;
+}
+
+const EditableKeyStatTile = ({
+  isEditing = false,
+  onRemove,
+  onSubmit,
+  dataBlockResponse: response,
+  dataBlockRequest,
+  id,
+  name,
+  summary,
+  ...props
+}: EditableKeyStatProps) => {
+  const [dataBlockResponse, setDataBlockResponse] = useState<
+    DataBlockResponse | undefined
+  >(response);
+  const [config, setConfig] = useState<KeyStatConfig>({
+    indicatorLabel: '',
+    value: '',
+  });
+  const [showForm, setShowForm] = useState<boolean>(false);
+  const [removing, setRemoving] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!dataBlockResponse) {
+      DataBlockService.getDataBlockForSubject(dataBlockRequest).then(
```

Comment:
Need to handle any API errors coming off this.  Look for examples of .catch(handleApiErrors) around the codebase

Reply:
Need to also add handleApiErrors to the dependency array for this effect